### PR TITLE
More arches (ppc64le + arm64v8) and ubuntu

### PR DESCRIPTION
**Pull request by yosifkit**
```diff
--- /dev/fd/63	2018-07-16 16:58:06.581643956 -0700
+++ /dev/fd/62	2018-07-16 16:58:06.578643898 -0700
@@ -1,25 +1,30 @@
-# this file is generated via https://github.com/docker-library/mariadb/blob/9ce4ecd2783d2358d2c0fe0efa924ecae33d9e84/generate-stackbrew-library.sh
+# this file is generated via https://github.com/docker-library/mariadb/blob/852044c20cd24a94bb15d8fcdb7b0784a40110d9/generate-stackbrew-library.sh
 
 Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon),
              Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
 GitRepo: https://github.com/docker-library/mariadb.git
 
-Tags: 10.3.8-jessie, 10.3-jessie, 10-jessie, jessie, 10.3.8, 10.3, 10, latest
-GitCommit: a7d1a184913c009c473baeb8f72385d12ae0de61
+Tags: 10.3.8-bionic, 10.3-bionic, 10-bionic, bionic, 10.3.8, 10.3, 10, latest
+Architectures: amd64, arm64v8, ppc64le
+GitCommit: 852044c20cd24a94bb15d8fcdb7b0784a40110d9
 Directory: 10.3
 
-Tags: 10.2.16-jessie, 10.2-jessie, 10.2.16, 10.2
-GitCommit: 1e7c5c9bb2bfde0453fb52175343a360ed346104
+Tags: 10.2.16-bionic, 10.2-bionic, 10.2.16, 10.2
+Architectures: amd64, arm64v8, i386, ppc64le
+GitCommit: 852044c20cd24a94bb15d8fcdb7b0784a40110d9
 Directory: 10.2
 
-Tags: 10.1.34-jessie, 10.1-jessie, 10.1.34, 10.1
-GitCommit: 12a0273332a3a0ed007ece2ae5886f7ac9c15d1b
+Tags: 10.1.34-bionic, 10.1-bionic, 10.1.34, 10.1
+Architectures: amd64, arm64v8, ppc64le
+GitCommit: 852044c20cd24a94bb15d8fcdb7b0784a40110d9
 Directory: 10.1
 
-Tags: 10.0.35-jessie, 10.0-jessie, 10.0.35, 10.0
-GitCommit: 20b1413b646df9a45df2d4dcc5c5c89ecb42c0b1
+Tags: 10.0.35-xenial, 10.0-xenial, 10.0.35, 10.0
+Architectures: amd64, arm64v8, i386, ppc64le
+GitCommit: 852044c20cd24a94bb15d8fcdb7b0784a40110d9
 Directory: 10.0
 
-Tags: 5.5.60-wheezy, 5.5-wheezy, 5-wheezy, 5.5.60, 5.5, 5
-GitCommit: 20b1413b646df9a45df2d4dcc5c5c89ecb42c0b1
+Tags: 5.5.60-trusty, 5.5-trusty, 5-trusty, 5.5.60, 5.5, 5
+Architectures: amd64, i386, ppc64le
+GitCommit: 852044c20cd24a94bb15d8fcdb7b0784a40110d9
 Directory: 5.5
```

Closes #175
Closes #158
Closes #155
Closes #111

### CODE REVIEW

**Comment by tianon** [comment]
Yay https://github.com/docker-library/php/pull/666 :tada: :cake:

**Comment by tianon** [comment]
Let's do this.

**Comment by grooverdan** [comment]
@yosifkit and @tianon thanks for much for making this happen.

File: 10.1/Dockerfile [hunk 1]
```diff
@@ -86,30 +104,34 @@ RUN echo "deb http://ftp.osuosl.org/pub/mariadb/repo/$MARIADB_MAJOR/debian jessi
...
-RUN { \
+RUN set -ex; \
+	{ \
 		echo "mariadb-server-$MARIADB_MAJOR" mysql-server/root_password password 'unused'; \
 		echo "mariadb-server-$MARIADB_MAJOR" mysql-server/root_password_again password 'unused'; \
-	} | debconf-set-selections \
-	&& apt-get update \
-	&& apt-get install -y \
+	} | debconf-set-selections; \
+	backupPackage='mariadb-backup-10.1'; \
```
[the remark this answers is not in the document]
Comment (tianon) [reply 10.1/Dockerfile]: I'm definitely sorry that your setup broke, but to say it was "without warning" is a little bit much.

We only have two ways of informing users of changes to images -- PRs/issues on this repository, and [PRs with the `library/mariadb` label on the official-images repository](https://github.com/docker-library/official-images/pulls?q=label%3Alibrary%2Fmariadb), of which this change had both, so there's unfortunately really not much more "heads-up" we could provide.
Additionally, I'd heartily recommend setting up some kind of testing environment where you can test image updates before deploying them to ensure that this sort of breakage doesn't affect you in the future.

All that being said, breaking folks wasn't an intention of this change, so it sounds like we need to do more to adapt this.  Can you provide more detail about what your setup looks like and how this broke it?  Ideally we'd love a couple simplified commands or a compose/stack YAML that can reproduce the failure.

**Comment by tianon** [comment]
Just to thread the needle, there were some minor issues with this on `arm64v8` which were fixed by https://github.com/docker-library/mariadb/pull/195.

File: 10.1/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (klausenbusk) [reply 10.1/Dockerfile]: > I'm definitely sorry that your setup broke, but to say it was "without warning" is a little bit much.

I'm sorry it came out a bit harsh.

> We only have two ways of informing users of changes to images -- PRs/issues on this repository, and PRs with the library/mariadb label on the official-images repository, of which this change had both, so there's unfortunately really not much more "heads-up" we could provide.


You are correct, but the change (xtrabackup -> mariabackup) should have been mentioned in the commit message (IMHO, its a breaking change).

> Additionally, I'd heartily recommend setting up some kind of testing environment where you can test image updates before deploying them to ensure that this sort of breakage doesn't affect you in the future.

It didn't break the cluster, it only broke bootstrapping of new database nodes.

> All that being said, breaking folks wasn't an intention of this change, so it sounds like we need to do more to adapt this. Can you provide more detail about what your setup looks like and how this broke it? Ideally we'd love a couple simplified commands or a compose/stack YAML that can reproduce the failure.

Galera use [SST](http://galeracluster.com/documentation-webpages/sst.html) for the initial node bootstrapping and you can choose between `mysqldump`, `rsync` or `xtrabackup`/`mariabackup`. So when `xtrabackup` was removed, you broke the `xtrabackup` SST method. It was easy to fix by switching to the `mariabackup` SST method.

[the remark this answers is not in the document]
Comment (yosifkit) [reply 10.1/Dockerfile]: @klausenbusk do you think that we should add a symlink for `xtrabackup` to `mariabackup` like you had in #158 or should we just leave it as is?

**Pull request merged**